### PR TITLE
fix: 🐛 DA-290 Replace signing identity with account identity while fetching account permissions

### PR DESCRIPTION
**Pull request by prashantasdeveloper**
### Description

Currently the signing Identity was used to get the permissions that an Account had as a Permissioned Account. This change replaces the use of `signingIdentity` with the corresponding identity associated with the Account and then uses that to fetch the permissions. Also, throws an error if the Account is not associated with any Identity

### Breaking Changes

🧨 `account.getPermissions` can throw if there is no Identity associated with the Account

### JIRA Link

DA-290

### Checklist

❌  Updated the Readme.md (if required) ?

### CODE REVIEW

File: src/api/entities/Account.ts [hunk 1]
```diff
@@ -470,21 +471,27 @@ export class Account extends Entity<UniqueIdentifiers, string> {
...
   public async getPermissions(): Promise<Permissions> {
-    const { context, address } = this;
+    const { address } = this;
+
+    const identity = await this.getIdentity();
 
-    const signingIdentity = await context.getSigningIdentity();
+    if (identity === null) {
+      throw new PolymeshError({
+        code: ErrorCode.DataUnavailable,
```
[the remark this answers is not in the document]
Comment (polymath-eric) [reply src/api/entities/Account.ts:484]: I think DataUnavailable is the right choice.

UnmetPrerequisite is more like creating an Asset without having a reserved ticker.

**Review by polymath-eric — COMMENTED** [review]
Code looks OK to me, but I don't understand why we need to make this change.

I thought `context.signingIdentity` would call `getIdentity` on the current account. Isn't this going to give the same behavior, or does this method get called outside of a signing context, that this fixes?

**Comment by prashantasdeveloper** [comment]
> Code looks OK to me, but I don't understand why we need to make this change.

Not always does one have a signingIdentity (like it would get called in REST API or even in case when SDK is initialized without a signingManager). `getPermissions` should be able to provide permissions irrespective whether there is signing Identity or not.

**Review by polymath-eric — APPROVED** [review]

File: src/api/entities/Account.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (monitz87) [reply src/api/entities/Account.ts:484]: Agree with eric, even though it's a close one

**Review by monitz87 — APPROVED** [review]

**Pull request merged**